mdagent/tools/base_tools/analysis_tools/thermo_tools.py

```diff
@@ -0,0 +1,335 @@
+        )
+
+
+class ComputeDipoleMoments(BaseTool):
+    name = "ComputeDipoleMoments"
+    description = """Compute the total dipole moment for each frame in a
+    molecular dynamics trajectory. Requires a trajectory file and a json file mapping
+    each atom in the trajectory to its partial charge.
+    Optionally requires a topology file.
+    Returns an array of dipole moments for each frame of the trajectory,
+    written to a file."""
+    path_registry: PathRegistry = PathRegistry().get_instance()
+
+    def __init__(self, path_registry: PathRegistry | None = None):
+        super().__init__()
+        if path_registry is not None:
+            self.path_registry = path_registry
+
+    def _compute_dipole_moments(self, traj: md.Trajectory, charges: np.ndarray):
+        return md.dipole_moments(traj, charges)
+
+    def _save_dipoles_to_file(self, dipole_moments: np.ndarray, traj_file: str):
+        file_name = self.path_registry.write_file_name(
+            type=FileType.RECORD,
+            record_type="dipole_moments",
+            file_format="csv",
+        )
+
+        save_path = f"{self.path_registry.ckpt_files}/{file_name}"
+        description = (
+            f"Dipole moments for each frame in the trajectory for file {traj_file}"
+        )
+
+        file_id = self.path_registry.get_fileid(
+            file_name=file_name, type=FileType.UNKNOWN
+        )
+        self.path_registry.map_path(file_id, save_path, description)
+
+        np.savetxt(
+            save_path,
+            dipole_moments,
+            delimiter=",",
+            header="Dipole_X,Dipole_Y,Dipole_Z",
+            comments="",
+        )
+        return file_id
+
+    def _run(self, traj_file: str, charge_file: str, top_file: str | None = None):
+        try:
+            traj = load_single_traj(
+                path_registry=self.path_registry,
+                traj_fileid=traj_file,
+                top_fileid=top_file,
+            )
+            if not traj:
+                raise Exception("Failed to load trajectory file.")
+        except Exception as e:
+            return str(e)
+
+        try:
+            charges = get_charges(self.path_registry, traj, charge_file)
+        except Exception as e:
+            return str(e)
+
+        dipole_moments = self._compute_dipole_moments(traj, charges)
+        file_id = self._save_dipoles_to_file(dipole_moments, traj_file)
+
+        return "Dipole moments computed and " f"saved to file {file_id}."
+
+    async def _arun(
+        self, traj_file: str, top_file: str | None = None, charges: np.ndarray = None
+    ):
+        raise NotImplementedError("Async version not implemented")
+
+
+class ComputeStaticDielectric(BaseTool):
+    name = "ComputeStaticDielectric"
+    description = """Compute the static dielectric constant of a system
+      from the dipole moments of a molecular dynamics trajectory.
+      Requires a trajectory file, charges, temperature and optionally a
```

Comment:
trajectory file ID

Reply:
Does the agent knows what arguments to use without describing them? particularly the top_file, which is the only kwarg

Reply:
I think I fixed this